### PR TITLE
CreateNewWiki: refactor starter import

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-1305

**TL;DR** - make databases and QA team happy with how fast CreateNewWiki works :)
### The current state

The way CreateNewWiki works currently is not the best possible solution. We basically generate on-the-fly dump of the selected starter wiki and import it to the newly created wiki. Once that's done we run  cleanup SQL queries that remove old article revisions. It all causes heavy load on database (queries to master, replication of many rows to all slaves). And it all takes time - the current average is ~30 sec (with spikes reaching over a minute).
### The proposed solution

On a daily basis generate two kind of dumps for each starter database (handled by `dumpStarters.php` script):
- content XML dump with only the recent revisions
- SQL dump with tables storing the relations between articles, templates, categories and images

These dumps are bzip2 compressed and stored on DFS:

```
$ s3cmd -c /etc/s3cmd/poz_dev.cfg ls s3://starter/dumps/
2015-07-14 10:05       998   s3://starter/dumps/aastarter.sql.bz2
2015-07-14 10:05      2892   s3://starter/dumps/aastarter.xml.bz2
2015-07-14 10:05      4657   s3://starter/dumps/destarter.sql.bz2
2015-07-14 10:05     24300   s3://starter/dumps/destarter.xml.bz2
2015-07-14 10:05      3638   s3://starter/dumps/esstarter.sql.bz2
2015-07-14 10:05     22100   s3://starter/dumps/esstarter.xml.bz2
...
```

Change CreateNewWiki controller to call `importStarter.php` maintenance script that imports these dumps (content first, followed by SQL dump) and updates the newly created wiki statistics (namely article count).

Tested these changes on a sandbox and the AJAX call now takes ~16 seconds.

@drozdo / @michalroszka / @wladekb / @Wikia/community-engineering

### CODE REVIEW

**Comment by wladekb** [comment]
lgtm

File: extensions/wikia/CreateNewWiki/classes/Starters.class.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+class Starters {
+
+	const STARTER_DUMPS_BUCKET = 'starter';
+
+	/**
+	 * starters map: langcode => database name
+	 *
+	 * "*" is default
+	 */
+	static private $mStarters = [
```
Comment (kamilkoterba) [inline extensions/wikia/CreateNewWiki/classes/Starters.class.php:21]: As you already touched this piece maybe you want to join it with https://github.com/Wikia/config/blob/dev/CommonExtensions.php#L2332
so we don't have duplicated code.

File: extensions/wikia/CreateNewWiki/maintenance/importStarter.php [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+			$this->logger->info(__METHOD__, [
+				'took' => microtime(true) - $then
+			]);
+		}
+		catch(Exception $ex) {
+			throw new CreateWikiException('runUpdates failed', self::ERR_RUN_UPDATES_FAILED, $ex);
+		}
+	}
+
+	public function execute() {
```
Comment (kamilkoterba) [inline extensions/wikia/CreateNewWiki/maintenance/importStarter.php:154]: Don't we have convention public methods on top?

Comment (macbre) [inline extensions/wikia/CreateNewWiki/maintenance/importStarter.php:154]: Never heard of it. Link please :)

File: extensions/wikia/CreateNewWiki/classes/Starters.class.php (hunk 1 above)
Comment (macbre) [inline extensions/wikia/CreateNewWiki/classes/Starters.class.php:21]: The entry for `it` is commented out in CommonExtensions

File: extensions/wikia/CreateNewWiki/maintenance/importStarter.php (hunk 2 above)
Comment (kamilkoterba) [inline extensions/wikia/CreateNewWiki/maintenance/importStarter.php:154]: verbal source: @Grunny :)
